### ISSUE TITLE
Using p4c-bm2-psa to compile a different architecture leads to a compiler bug

### ISSUE THREAD

**Issue by fruffy** [p4lang/p4c#2223]
Very simple to reproduce, an example is this command:
`p4c/build/p4c-bm2-psa p4c/testdata/p4_16_samples/key-bmv2.p4`

```bash
[--Wwarn=invalid] warning: V1Switch: the main package should be called PSA_Switch; are you using the wrong architecture?                    
package V1Switch<H, M>(Parser<H, M> p,
        ^^^^^^^^
In file: /mnt/windows/Projekte/p4_tv/p4c/ir/ir.cpp:219
Compiler Bug: No parameter named ingress
```

**Comment by jnfoster** [comment]
Thanks. Though PSA support is sub-alpha quality. I wonder if it's premature to start fuzzing it...

**Comment by fruffy** [comment]
Oh this was not even due to fuzzing, I just accidentally switched binaries while compiling a file ha.
But that is good to know though!

**Comment by jnfoster** [comment]
I mean, this target definitely needs tested. But more work will happen on the PSA backend soon, so maybe we should wait until there is a more viable binary to validate :-)

**Comment by fruffy** [comment]
This was more of an accident which we filed for completeness sake. We are not planning to target PSA any time soon. But I didn't know that the psa-backend is still sub-alpha quality so that is pretty useful to keep in mind for the future.

**Comment by mbudiu-vmw** [comment]
Fixed by #2531
